bork/builder.py

```diff
@@ -25,10 +25,8 @@ def dist():
 
 
 def version_from_sdist_file():
-    sdist = sorted(Path.cwd().glob('dist/*.tar.gz'))[-1].name
-    sdist = sdist.replace('.tar.gz', '')
-    sdist = sdist.split('-')[1]
-    return sdist
+    sdist = max(Path.cwd().glob('dist/*.tar.gz')).name
```

Comment:
This likely ought to be a more-precise glob, rather than relying on the sdist's name being the last one (in lexicographic order).  However I did not want to make a change that didn't preserve functional equivalence, given that this isn't documented.

Reply:
@nbraud honestly, using the sdist files at all is a _huge_ kludge because I couldn't find a better way to get that information. The only place `version_from_sdist_file()` is used is in `bork.builder.zipapp()`. I'll open an issue for improving how version information is found.

Also, this `BAD ASSUMPTION` comment is caused entirely by the way this function works: https://github.com/duckinator/bork/blob/master/bork/builder.py#L60

Reply:
I've looked into that for GoBuild.it. If you assume setuptools, there's some commands to produce the metadata directories (`dist_info` and `egg_info`). If you're assuming PEP517, there aren't universal ways to get the metadata about a project--you have to build a distribution first.